### ISSUE TITLE
Bug: GPS / Use My Location button is disabled on map

### ISSUE THREAD

**Issue by fancyham** [hackforla/food-oasis#1101]
### Overview
In the past few days, the ‘crosshairs’ button to get the user’s location via GPS has been disabled both on web and mobile when in map mode.

Note the “Use my current location” button on landing page does work properly.

Perhaps this is a regression due to a recent update?

### How to replicate
1. Go to foodoasis.la
2. Tap magnifying glass to show all of LA
3. Notice that crosshairs button in upper right corner is greyed out:

<img width="320" alt="image" src="https://user-images.githubusercontent.com/3376957/142693406-5ffde934-17fe-486e-882d-1a6824eab4c8.png">

### Platforms and apps
Safari desktop
Safari mobile

### Action Items
- [ ] Reactivate ‘get my location’ button

### Resources/Examples

**Comment by entrotech** [comment]
This is now implemented as follows:
If the user has not enabled location for the site in their browser, the "GPS" (aka re-center) button will be disabled.
If the user has enabled location for the site in their browser, then when the re-center button is pressed, the map will be re-centered on the user's location, and the "start" pin, if any, will be moved to the user's location as well.

**Comment by fancyham** [comment]
I think I saw an issue where even when I have granted access to my location, the GPS button looks disabled.  In the screenshot below, I’ve granted access to my location when loading the site.  The white ‘target’ icon on light grey as shown here looks disabled to me, but if I roll over it, the button background turns blue, but can we instead can instead blue target rather than white?  And only use the white target if the GPS is not working and truly disabled
![Screen Shot 2022-03-24 at 7 35 20 PM](https://user-images.githubusercontent.com/3376957/160044515-cf8b659a-9c40-4e82-918e-0bf84077707f.jpg)

Also, possible bug?

- [ ] It seems like if I deny access to my location when first loading the site, then grant access, the site still thinks that I have denied it permanently and the GPS button doesn’t work?

Perhaps access to the browser’s location is stored in the user’s session and if denied once, subsequent requests are not made?  If so, can we change that to allow users to turn on or allow location on subsequent requests?

- [ ] If possible, I’d love to see this sequence, which is a possible change:

If we know that GPS is disabled, then disable the GPS button… but also check periodically if the user has re-enabled location sharing in their browser (as I tried above) and reenable button as appropriate.

Or: 

Always make GPS button ‘active’ (blue target on the button) and request the GPS location every time the user presses the button… then show a message like “could not get location from browser” when that request fails?

Also:

- [ ] Can we change the button tooltip from ‘re-center’ to ’show your location’ (Google calls it “show your location”, Apple uses “show your current location")

**Comment by fancyham** [comment]
I was using the wrong URL!

Looks good — one problem I came across:
- [ ] It seems like if I deny access to my location when first loading the site, then grant access, the site still thinks that I have denied it permanently and the GPS button doesn’t work?  The button still looks clickable (blue background) but it doesn’t seem to do anything.

To recreate:
1. Mac Safari (Monterey, latest version of Safari)

2. Click ‘Use my current location’, but do NOT select ‘remember for a day"

3. When Safari asks, deny access to location
4. Type in ‘LA’ and get map 

5.GPS button looks active.  Click it.
6. Nothing happens.   

Expectation is that the website would request my location again and I would see a Safari location request dialog again. 



- [ ] Can we change the button tooltip from ‘re-center’ to ’show your location’ (Google calls it “show your location”, Apple uses “show your current location”).  This would be a better description than ‘re-center’.
